_posts/2025-06-17-rios-ecs.markdown

```diff
@@ -0,0 +1,95 @@
+
+[RIOS supports ECS](https://www.rioshome.org/en/latest/concurrency.html#CW_ECS) in 2 modes - Fargate and Private Cluster. Examples for both of these 
+are below. Note that the older AWS Batch parallel support is deprecated - ECS is a much
+neater solution. You can still run the "main" RIOS script from within AWS Batch of course.
+
+## ECS with Fargate
+
+Fargate is a mode of ECS that is much easier to use - AWS has a fleet of VMs that are already
+running and can be added to your ECS cluster with the mimimum of hassle (and speed). Here is an exampe
+of how you might start and ECS Fargate cluster using the helper function [makeExtraParams_Fargate](https://www.rioshome.org/en/latest/rios_computemanager.html#rios.computemanager.ECSComputeWorkerMgr.makeExtraParams_Fargate):
+
+```
+from rios import applier
+from rios.computemanager import ECSComputeWorkerMgr
+...
+extra_params = ECSComputeWorkerMgr.makeExtraParams_Fargate(
+    jobName='myjobname',
+    subnets=mysubnets,
+    containerImage=MyECRImage,
+    taskRoleArn=MyECSTaskRoleARN,
+    executionRoleArn=MyECSTaskExecutionRoleARN,
+    securityGroups=MySecurityGroup,
+    cpu='2 vCPU', memory='16GB', cpuArchitecture='ARM64')
+
+conc = applier.ConcurrencyStyle(
+            computeWorkerKind=applier.CW_ECS,
+            computeWorkerExtraParams=extra_params,
+            ...)
+controls.setConcurrencyStyle(conc)
+```
+
+Note how you can adjust memory, CPU and CPU Architecture. See [makeExtraParams_Fargate](https://www.rioshome.org/en/latest/rios_computemanager.html#rios.computemanager.ECSComputeWorkerMgr.makeExtraParams_Fargate) for a more thorough description of these options.
+Note that you may well need to create these resources before hand with CloudFormation/CDK/Terraform. A future blog post may cover this in more detail.
+
+# ECS with Private Cluster
+
+Setting up an ECS Private Cluster is more work, but gives you far more control over the kind of
+CPU that can be chosen (GPU etc) and you can even control the AMI that the VMs will run. However, it will be slower to start. Below is an example of setting up a Private Cluster with the helper function  [makeExtraParams_PrivateCluster](https://www.rioshome.org/en/latest/rios_computemanager.html#rios.computemanager.ECSComputeWorkerMgr.makeExtraParams_PrivateCluster):
+
+```
+from rios import applier
+from rios.computemanager import ECSComputeWorkerMgr
+...
+extra_params = ECSComputeWorkerMgr.makeExtraParams_PrivateCluster(
+            jobName='myjobname`,
+            numInstances=computeworkers,
+            ami='ami-08b5d886b9289bc63',
+            instanceType='c7g.2xlarge',
+            containerImage=MyECRImage,
+            taskRoleArn=MyECSTaskRoleARN,
+            executionRoleArn=MyECSTaskExecutionRoleARN,
+            subnet=subnets[0],
+            securityGroups=MySecurityGroup,
+            instanceProfileArn=MyInstanceProfileARN,
+            memoryReservation=16384)
```

Comment:
Possibly leave out the memoryReservation=16384. After your earlier difficulties with setting that, I changed to docstring to suggest leaving it at default until a memory error required that it be increased.